### PR TITLE
 maks sure we don't change dtype in zero_mean preprocessing

### PR DESCRIPTION
**Pull request by FynnBe**

### CODE REVIEW

File: tiktorch/runner/prediction_pipeline/_preprocessing.py [hunk 1]
```diff
@@ -32,7 +32,11 @@ def zero_mean_unit_variance(tensor: xr.DataArray, axes=None, eps=1.0e-6, mode="p
     if mode != "per_sample":
         raise NotImplementedError(f"Unsupported mode for zero_mean_unit_variance: {mode}")
 
-    return (tensor - mean) / (std + 1.0e-6)
+    ret = (tensor - mean) / (std + 1.0e-6)
+
+    # monkey patch: maks sure we don't change dtype
+    # todo: allow preprocessing to change dtype?
+    return ret.astype(tensor.dtype)
```
Comment (constantinpape) [inline tiktorch/runner/prediction_pipeline/_preprocessing.py:39]: Wouldn't it make more sense to always cast to float32? Going back to earlier discussions, we said that models always take float32.

Comment (FynnBe) [reply tiktorch/runner/prediction_pipeline/_preprocessing.py:39]: yes, although in-between preprocessing steps it also may differ from float32... 
It's a monkey patch either way. I wouldn't be oposed to hard-code float32 here (in fact that was my first uncommited version)

**Comment by FynnBe** [comment]
I'm having trouble trying to test this fix with this model: https://bioimage.io/#/?id=10.5072%2Fzenodo.880412
It seems like my pytorch version does not match or something similar environment related:
```
C:\conda\envs\tiktorch-server-env\python.exe C:/repos/tiktorch/tiktorch/runner/model_test.py -m C:\Users\fbeut\Downloads\em-mitochondria-boundarymodel_pytorch_state_dict\rdf.yaml
Traceback (most recent call last):
  File "C:/repos/tiktorch/tiktorch/runner/model_test.py", line 51, in model_test
    model_zip, devices=["cuda" if torch.cuda.is_available() else "cpu"], preserve_batch_dim=True
  File "C:\repos\tiktorch\tiktorch\runner\utils.py", line 37, in eval_model_zip
    bioimageio_model=bioimageio_model, devices=devices, preserve_batch_dim=preserve_batch_dim
  File "C:\repos\tiktorch\tiktorch\runner\prediction_pipeline\_prediction_pipeline.py", line 201, in create_prediction_pipeline
    bioimageio_model=bioimageio_model, devices=devices, weight_format=weight_format
  File "C:\repos\tiktorch\tiktorch\runner\prediction_pipeline\_model_adapters\_model_adapter.py", line 73, in create_model_adapter
Failed to load prediction pipeline from source package
    return adapter_cls(bioimageio_model=bioimageio_model, devices=devices)
  File "C:\repos\tiktorch\tiktorch\runner\prediction_pipeline\_model_adapters\_pytorch_model_adapter.py", line 24, in __init__
    state = torch.load(weights.source, map_location=self.devices[0])
  File "C:\conda\envs\tiktorch-server-env\lib\site-packages\torch\serialization.py", line 367, in load
    return _load(f, map_location, pickle_module)
  File "C:\conda\envs\tiktorch-server-env\lib\site-packages\torch\serialization.py", line 528, in _load
    magic_number = pickle_module.load(f)
_pickle.UnpicklingError: A load persistent id instruction was encountered,
but no persistent_load function was specified.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:/repos/tiktorch/tiktorch/runner/model_test.py", line 98, in <module>
    sys.exit(main())
  File "C:/repos/tiktorch/tiktorch/runner/model_test.py", line 90, in main
    raise e
  File "C:/repos/tiktorch/tiktorch/runner/model_test.py", line 87, in main
    model_test(args.model, args.weight_format, args.decimals)
  File "C:/repos/tiktorch/tiktorch/runner/model_test.py", line 54, in model_test
    raise RuntimeError("Failed to load prediction pipeline from source package") from e
RuntimeError: Failed to load prediction pipeline from source package

Process finished with exit code 1
```

File: tiktorch/runner/prediction_pipeline/_preprocessing.py (hunk 1 above)
Comment (constantinpape) [reply tiktorch/runner/prediction_pipeline/_preprocessing.py:39]: Ok, let's go with float32 hard-coded for now.

**Comment by constantinpape** [comment]
What pytorch version do you have? It should be >1.6.

**Comment by FynnBe** [comment]
> What pytorch version do you have? It should be >1.6.

1.9.0

File: tiktorch/runner/prediction_pipeline/_preprocessing.py (hunk 1 above)
Comment (constantinpape) [inline tiktorch/runner/prediction_pipeline/_preprocessing.py:39]:
```suggestion
    return ret.astype("float32")
```

**Comment by k-dominik** [comment]
pretty late to the game, but the patch will fail for any kind of integer data.

**Comment by constantinpape** [comment]
> pretty late to the game, but the patch will fail for any kind of integer data.

I see. It was merged in a bit of a panic mode...
Let's discuss a clean solution in #184.

**Pull request merged**